ID: 13017
Title: duckdb.duckdb.IOException: IO Error: Could not set lock on "/Volumes/homefolder/test1.db" Operation Not Supported
Description:
### What happens?

I have a db stored on a samba partition. I can't connect to it even in read_only mode.

This is the error I get:
`duckdb.duckdb.IOException: IO Error: Could not set lock on file "/Volumes/homefolder/test1.db": Operation not supported. See also https://duckdb.org/docs/connect/concurrency`



### To Reproduce

```python
import duckdb

my_database = "/Volumes/homefolder/test1.db"
con = duckdb.connect()
#con.install_extension("fts")
conn = duckdb.connect(my_database, read_only=True)
```

### OS:

aarch64

### DuckDB Version:

1.0.0

### DuckDB Client:

Python

### Full Name:

Tommaso Bendinelli

### Affiliation:

CSEM

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have